cost/volumes/uav_policy.pt

```diff
@@ -0,0 +1,288 @@
+
+    // Create a Date object based on the yyyy, mm, dd
+    // NOTE: months in javascript goes 0-11 so subtract one from the mm bit 
+    var updated_at_date = new Date(date_parts[0], (Number(date_parts[1]) - 1), date_parts[2])
+    var updated_at_ms = updated_at_date.getTime()
+    var difference_ms = Math.abs(today_ms - updated_at_ms)
+    var number_of_days_old = Math.round(difference_ms/ONE_DAY)
+
+    // Add a computed days old item to the hash
+    my_vol["age"] = number_of_days_old
+
+    // Push the volume hash element to the unattach_volumes array
+    unattached_volumes.push(my_vol)  
+  }
+}
+EOS
+end
+	
+policy "pol_unattached_volumes" do
+  validate_each $ds_unattached_volumes do
+	# Go through the unattached volumes that were found by the script-based datasource above
+	# and check if not older than specified number of days.
+	# If the check fails, it'll flag the volume
+	check le(val(item, "age"), $param_days_old)
+
+	escalate $esc_unattached_volumes
+
+	summary_template "Unattached Volumes"
+
+#    	detail_template <<-EOS
+#Unattached Volume Report for Account: {{ rs_project_name }} (ID: {{ rs_project_id }})
+#The following unattached volumes have exceeded the specified age of: {{ parameters.param_days_old }} days old
+#{{ range data }}
+#ID: {{ .id }}
+#Cloud HREF: {{ .cloud_href }}
+#Name: {{ .name }}
+#Size: {{ .size }}
+#Status: {{ .status }}
+#Azure Disk Type: {{ .azure_disk_type }}
+#Azure Storage Type: {{ .azure_storage_type }}
+#Azure Storage Account Name: {{ .azure_storage_account_name }}
+#Unattached on: {{ .updated_at }}
+#Current Age (days): {{ .age}}
+#{{ end }}
+#EOS
+
+
+  detail_template <<-EOS
+Unattached Volume Report for Account: {{ rs_project_name }} (ID: {{ rs_project_id }})\n
+
+The following unattached volumes have exceeded the specified age of: {{ parameters.param_days_old }} days old\n
+| Cloud Name | Volume Name | Age (days) | Volume Type | Resource UID | Size (GBs) | Status | IOPS | Azure Disk Type | Azure Storage Type | Azure Storage Account | Created At | Updated At | Volume HREF |
+| ---------- | ----------- | --- | ----------- | ------------ | ---- | ------ | ---- | --------------- | ------------------ | --------------------- | ---------- | ---------- | ----------- |
+{{ range data -}}
+| {{ .cloud_name }} | {{ .name }} | {{ .age }} | {{ .volume_type }} | {{ .id }} | {{ .size }} | {{ .status }} | {{ .iops }} | {{ .azure_disk_type }} | {{ .azure_storage_type }} | {{ .azure_storage_account_name }} | {{ .created_at }} | {{ .updated_at }} | {{ .href }} |
+{{ end -}}
+EOS
+
+  end
+end
+
+
+escalation "esc_unattached_volumes" do
+  email $param_email
+  
+  run "delete_volumes", data
+end
+
+define delete_volumes($data) do
+  
+  foreach $item in $data do
+    if $item["status"] == "DELETED"
+      rs_cm.audit_entries.create(notify: "None", audit_entry: { auditee_href: @@account, summary: "Policy Deletion of Volume, "+$item["name"] , detail: to_s($item)})
```

Comment:
should the policy created an audit entry for each item.  I beleive volume create and delete are always audited.   the extra api calls cause the policy to take longer to complete.